src/App.tsx

```diff
@@ -1,223 +1,223 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useEffect, useRef, useState, useContext, useCallback, FormEvent, useMemo,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { FilterTypes } from './types/FilterType';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { TodoList } from './components/TodoList';
+import {
+  getTodos,
+  createTodo,
+  deleteTodos,
+  updateTodo,
+} from './api/todos';
+
+export function getFilterTodos(
+  todos: Todo[],
+  filterTypes: FilterTypes,
+) {
+  const filterTodo = [...todos];
+
+  switch (filterTypes) {
+    case FilterTypes.Active:
+      return filterTodo.filter(({ completed }) => !completed);
+
+    case FilterTypes.Completed:
+      return filterTodo.filter(({ completed }) => completed);
+
+    default:
+      return filterTodo;
+  }
+}
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const user = useContext(AuthContext);
+  const user = useContext<User | null>(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterTypes>(FilterTypes.All);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [isAdding, setAdding] = useState(false);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    const todoFromServer = async (userId: number) => {
+      try {
+        const todosFromServer = await getTodos(userId);
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setErrorMessage(`${error}`);
+      }
+    };
+
+    if (!user) {
+      return;
     }
+
+    todoFromServer(user.id);
   }, []);
 
+  const addNewTodo = useCallback(async (event:FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim() || !user) {
+      setErrorMessage('Title can\'t be empty');
+
+      return;
+    }
+
+    setAdding(true);
+
+    try {
+      const newTodo = await createTodo(user.id, title);
+
+      setTodos(prevTodo => ([...prevTodo, newTodo]));
+    } catch {
+      setErrorMessage('Unable to add a todo');
```

Comment:
you could create enum for error messages, it's a good practice not to rely on the unknown string